### PR TITLE
docs: Elaborate on how the `php_fastcgi` directive works

### PR DESCRIPTION
**Pull request by francislavoie**
This is mostly taken from a post I wrote on the forums last month https://caddy.community/t/php-fastcgi-only-serving-php-files-or-wildcard-possible/13918/3 which I thought was a good and thorough explanation of how the directive works, and why it works that way. Did a bit of polish on the wording the fit the language the docs usually use (IMO anyways).

Also added a ToC since the page gets a bit long after these additions, best to give users a way to jump down to the examples quickly.

### CODE REVIEW

File: src/docs/markdown/caddyfile/directives/php_fastcgi.md [hunk 1]
```diff
@@ -49,7 +54,7 @@ Since this directive is an opinionated wrapper over a reverse proxy, you can use
 
 ## Expanded form
 
-The `php_fastcgi` directive is the same as the following configuration:
+The `php_fastcgi` directive (with none of the subdirectives configured) is the same as the following configuration. Most modern PHP apps work well with this preset. If yours does not, feel free to borrow from this and customize it as needed instead of using the `php_fastcgi` shortcut.
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/php_fastcgi.md]:
```suggestion
The `php_fastcgi` directive (without subdirectives) is the same as the following configuration. Most modern PHP apps work well with this preset. If yours does not, feel free to borrow from this and customize it as needed instead of using the `php_fastcgi` shortcut.
```

File: src/docs/markdown/caddyfile/directives/php_fastcgi.md [hunk 2]
```diff
@@ -77,7 +82,27 @@ route {
 }
 ```
 
-Most modern PHP apps work well with this preset. If yours does not, feel free to borrow from this and customize it as needed instead of using the `php_fastcgi` shortcut.
+### Explanation
+
+Since the above expanded form may be hard to follow, here's an in-depth explanation of what it directive does, and why it works this way.
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/php_fastcgi.md]:
```suggestion
```

(Remove fluff, an explanation is great, no need to justify it IMO)

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for this, people like explanations. Just a minor nit.

**Review by mholt — APPROVED** [review]

**Comment by MestreLion** [comment]
That's a great addition!

Could also mention that some in systems, such as Debian and Ubuntu, `php-fpm` by default listens on a _UNIX (file) socket_, not a network address. So the example `php_fastcgi http://127.0.0.1:9000` should be changed to `php_fastcgi unix//run/php/php7.4-fpm.sock` in such systems
